de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/impl/OLCandBL.java

```diff
@@ -216,17 +218,20 @@ public InvoiceRule getInvoiceRule(@Nullable final BPartnerOrderParams bPartnerOr
 	}
 
 	@Override
-	public PaymentRule getPaymentRule(@Nullable final BPartnerOrderParams bPartnerOrderParams, @Nullable final OLCandOrderDefaults orderDefaults)
+	public PaymentRule getPaymentRule(@Nullable final BPartnerOrderParams bPartnerOrderParams,
+			@Nullable final OLCandOrderDefaults orderDefaults,
+			@Nullable I_C_OLCand orderCandidateRecord)
 	{
-		if (bPartnerOrderParams != null)
-		{
-			return bPartnerOrderParams.getPaymentRule();
-		}
-		if (orderDefaults != null)
-		{
-			return orderDefaults.getPaymentRule();
-		}
-		return null;
+		final PaymentRule orderCandidatePaymentRule = orderCandidateRecord == null ? null
+				: PaymentRule.ofCode(orderCandidateRecord.getPaymentRule());
```

Comment:
as above: `PaymentRule` is not mandatory, so it might be `null`

Reply:
Used ofNullableCode instead.